test/e2e_test_cluster_image_policy_with_source.sh

```diff
@@ -0,0 +1,132 @@
+    exit 1
+  else
+    export OIDC_TOKEN=`curl -s ${ISSUER_URL}`
+  fi
+fi
+
+if [[ -z "${KO_DOCKER_REPO}" ]]; then
+  echo "Must specify env variable KO_DOCKER_REPO"
+  exit 1
+fi
+
+if [[ -z "${FULCIO_URL}" ]]; then
+  echo "Must specify env variable FULCIO_URL"
+  exit 1
+fi
+
+if [[ -z "${REKOR_URL}" ]]; then
+  echo "Must specify env variable REKOR_URL"
+  exit 1
+fi
+
+if [[ -z "${TUF_ROOT_FILE}" ]]; then
+  echo "must specify env variable TUF_ROOT_FILE"
+  exit 1
+fi
+
+if [[ -z "${TUF_MIRROR}" ]]; then
+  echo "must specify env variable TUF_MIRROR"
+  exit 1
+fi
+
+if [[ "${NON_REPRODUCIBLE}"=="1" ]]; then
+  echo "creating non-reproducible build by adding a timestamp"
+  export TIMESTAMP=`date +%s`
+else
+  export TIMESTAMP="TIMESTAMP"
+fi
+
+# Initialize cosign with our TUF root
+cosign initialize --mirror ${TUF_MIRROR} --root ${TUF_ROOT_FILE}
+
+# Publish the first test image
+echo '::group:: publish test image demoimage'
+pushd $(mktemp -d)
+go mod init example.com/demo
+cat <<EOF > main.go
+package main
+import "fmt"
+func main() {
+  fmt.Println("hello world TIMESTAMP")
+}
+EOF
+
+sed -i'' -e "s@TIMESTAMP@${TIMESTAMP}@g" main.go
+cat main.go
+export demoimage=`ko publish -B example.com/demo`
+echo Created image $demoimage
+popd
+echo '::endgroup::'
+
+echo '::group:: Deploy ClusterImagePolicy with keyless signing and source'
+kubectl apply -f ./test/testdata/policy-controller/e2e/cip-keyless-with-source.yaml
+echo '::endgroup::'
+
+echo '::group:: Sign demo image'
+COSIGN_EXPERIMENTAL=1 cosign sign --rekor-url ${REKOR_URL} --fulcio-url ${FULCIO_URL} --force --allow-insecure-registry ${demoimage} --identity-token ${OIDC_TOKEN}
+echo '::endgroup::'
+
+echo '::group:: Verify demo image'
+COSIGN_EXPERIMENTAL=1 cosign verify --rekor-url ${REKOR_URL} --allow-insecure-registry ${demoimage}
+echo '::endgroup::'
+
+echo '::group:: Create test namespace and label for verification'
+kubectl create namespace demo-keyless-source
+kubectl label namespace demo-keyless-source policy.sigstore.dev/include=true
+export NS=demo-keyless-source
+echo '::endgroup::'
+
+echo '::group:: test job success as source.oci points to the right repository'
+# We signed this above, this should work
+if ! kubectl create -n demo-keyless-source job demo --image=${demoimage} ; then
+  echo Failed to create Job in namespace with matching signature and good OCI source!
+  exit 1
+else
+  echo Succcessfully created Job with signed image and good OCI source
+fi
+echo '::endgroup::'
+
+echo '::group:: Change source.oci in policy to a wrong repository without signatures'
+kubectl patch cip image-policy-keyless-source --type "json" \
+-p '[{"op":"replace", "path":"/spec/authorities/0/source/0/oci", "value":"ghcr.io/sigstore/cosign/cosign"}]'
+# allow for propagation
+sleep 5
+echo '::endgroup::'
+
+echo '::group:: test job success as source.oci points to the right repository'
+# We signed this above, this should work
```

Comment:
nit, this comment is wrong? This should fail because the it's trying to fetch signatures that don't exist.